dotenv/src/parse.rs

```diff
@@ -1,63 +1,57 @@
 use std::collections::HashMap;
 
-use lazy_static::lazy_static;
-use regex::{Captures, Regex};
-
 use crate::errors::*;
 
 // for readability's sake
 pub type ParsedLine = Result<Option<(String, String)>>;
 
-pub fn parse_line(line: &str, mut substitution_data: &mut HashMap<String, Option<String>>) -> ParsedLine {
-    lazy_static! {
-      static ref LINE_REGEX: Regex = Regex::new(r#"(?x)
-        ^(
-          \s*
-          (
-            \#.*|                             # A comment, or...
-            \s*|                              # ...an empty string, or...
-            (export\s+)?                      # ...(optionally preceded by "export")...
-            (?P<key>[A-Za-z_][A-Za-z0-9_.]*)  # ...a key,...
-            =                                 # ...then an equal sign,...
-            (?P<value>.+?)?                   # ...and then its corresponding value.
-          )\s*
-        )
-        [\r\n]*
-        $
-      "#).unwrap();
+pub fn parse_line(
+    original_line: &str,
+    mut substitution_data: &mut HashMap<String, Option<String>>,
+) -> ParsedLine {
+    // trim whitespace
+    let mut line = original_line.trim();
+
+    // if its an empty line or a comment, skip it
+    if line.is_empty() || line.starts_with('#') {
+        return Ok(None);
     }
 
-    LINE_REGEX
-        .captures(line)
-        .map_or(Err(Error::LineParse(line.into(), 0)), |captures| {
-            let key = named_string(&captures, "key");
-            let value = named_string(&captures, "value");
+    // optionally preceded by `export\s+`
+    if line.starts_with("export") {
+        let without_export = line.trim_start_matches("export");
+        if !without_export.starts_with(|c: char| c.is_whitespace()) {
+            return Err(Error::LineParse(original_line.into(), 0));
```

Comment:
Now that we have a manual parser we should calculate the index of the parsing error here. Make sure to update the rest of the places that return errors as well.